### PR TITLE
feat: add upload rate config support

### PR DESCRIPTION
**Pull request by rungao**
By default, data is uploaded every 10 seconds, but sometimes you need to customize the upload frequency to control the number of data collection.

### CODE REVIEW

**Comment by eh-am** [comment]
Just out of curiosity, can you tell me more about the use case? Do you need to run it more or less often?

File: pyroscope/api.go [hunk 1]
```diff
@@ -22,6 +22,7 @@ type Config struct {
 	DisableAutomaticResets bool // disable automatic profiler reset every 10 seconds. Reset manually by calling Flush method
 	// Deprecated: the field is ignored and does nothing
 	DisableCumulativeMerge bool
+	UploadRate             uint8 // data upload frequency: every 10 seconds by default
```
Comment (kolesnikovae) [inline pyroscope/api.go:25]: Please consider using `time.Duration` type

**Comment by rungao** [comment]
> Just out of curiosity, can you tell me more about the use case? Do you need to run it more or less often?

There are two factors that I want to adjust the upload frequency :
1. We connected to over 20 apps that taking up too much storage. 
image

2. Sometimes all data is lost  and  cannot to upload data. I suspect it has something to do with uploading too much data.

File: pyroscope/api.go (hunk 1 above)
Comment (rungao) [reply pyroscope/api.go]: https://github.com/pyroscope-io/client/pull/31/commits/f72f6e75cafa57a14fd85f6cacf38c5b2abf47a8

**Comment by petethepig** [comment]
Duplicate with #32, closing

**Pull request closed without merging**